accounting/third_party_api/google/tests/test_client.py

```diff
@@ -20,3 +20,44 @@
 """
 Tests for the  Google API client.
 """
+
+from unittest import mock
+
+from accounting.common.tests.base import TestCase
+from accounting.third_party_api.google.client import GoogleDrive
+
+
+class GoogleDriveTestCase(TestCase):
+    """
+    Test cases for the Google Drive client.
+    """
+
+    def setUp(self):
+        """Set up test objects."""
+        super().setUp()
+        self.google_drive = GoogleDrive()
+        self.path = ['non-existant-folder', '2018', 'invoices-in', '01']
+
+        # Set up mocks.
+        get_list_mock = mock.patch('pydrive.apiattr.ApiResourceList.GetList')
+        self.get_list_mock = get_list_mock.start()
+        self.addCleanup(get_list_mock.stop)
+
+    def test_get_folder_id(self):
+        """The ID of the folder at the end of the path."""
+        self.get_list_mock.side_effect = [
+            [],
```

Comment:
@UmanShahzad If this list is supposed to contain a path matching `self.path` above, shouldn't the first item look like this:

```python
            [{'title': 'non-existent-folder', 'id': None}]
```

Reply:
@itsjeyd Actually, this is to meet the case when `folders` in the function itself is empty. So why I put the name 'non-existent-folder' is to signal that the folder doesn't exist in GDrive and nothing will be returned when trying to look it up.